app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,68 @@
+class MarketsController < ApplicationController
+
+  before_action :get_market, only: [:edit, :update]
+
+  def get_market
+    @market = Market.find(params[:id])
+  end
+
+  def index
```

Comment:
If you change index to a blank method, move the index.html.erb from app/views/users/ to app/views/markets, then you can create a new method with the code beneath this (such as list) and call that to list all markets. Then change the routes.rb to have:  get  "/all_markets"        => "markets#list".  At that point you should be able to remove the users_controller.rb, the users model, and the remaining users views.